crates/proof-of-sql/src/sql/postprocessing/group_by_postprocessing.rs

```diff
@@ -169,6 +177,153 @@ impl GroupByPostprocessing {
+        let group_by_ins = self
+            .group_by_identifiers
+            .clone()
+            .into_iter()
+            .map(|id| {
+                let column = owned_table
+                    .inner_table()
+                    .get(&id)
+                    .ok_or(PostprocessingError::ColumnNotFound(id.to_string()))?;
+                Ok(Column::<S>::from_owned_column(column, &alloc))
+            })
+            .collect::<PostprocessingResult<Vec<_>>>()?;
+        // TODO: Allow a filter
+        let selection_in = vec![true; owned_table.num_rows()];
+        let (sum_ids, sum_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Sum)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let (max_ids, max_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Max)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let (min_ids, min_ins): (Vec<_>, Vec<_>) = evaluated_columns
+            .get(&AggregationOperator::Min)
+            .map(|tuple| {
+                tuple
+                    .iter()
+                    .map(|(id, c)| (*id, Column::<S>::from_owned_column(c, &alloc)))
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let count_ids: Vec<_> = evaluated_columns
+            .get(&AggregationOperator::Count)
+            .unwrap_or(&vec![])
+            .iter()
+            .map(|(id, _)| *id)
+            .collect::<Vec<_>>();
+        let aggregation_results = aggregate_columns(
+            &alloc,
+            &group_by_ins,
+            &sum_ins,
+            &max_ins,
+            &min_ins,
+            &selection_in,
+        )?;
+        // Finally do another round of evaluation to get the final result
+        // Gather the results into a new OwnedTable
+        let group_by_outs: IndexMap<Identifier, OwnedColumn<S>> = aggregation_results
+            .group_by_columns
+            .iter()
+            .zip(self.group_by_identifiers.iter())
+            .map(|(column, id)| (*id, OwnedColumn::from(column)))
+            .collect::<IndexMap<_, _>>();
+        let sum_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.sum_columns.iter(),
+            sum_ids.iter(),
+            sum_ins.iter(),
+        ))
+        .map(|(c_out, id, c_in)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_scalars(c_out, c_in.column_type())?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
```

Comment:
See other ⭐ comments.
```suggestion
        });
```